libs/wal_decoder/src/serialized_batch.rs

```diff
@@ -541,6 +541,25 @@ impl SerializedValueBatch {
         })
     }
 
+    /// Generates batch statistics for metrics.
+    pub fn stats(&self) -> SerializedValueStats {
+        let mut stats = SerializedValueStats::default();
+        for metadata in &self.metadata {
+            let ValueMeta::Serialized(valuemeta) = metadata else {
+                continue;
+            };
+            let is_block_key = Key::from_compact(valuemeta.key).is_block_key();
+            let will_init = valuemeta.will_init;
+            match (is_block_key, will_init) {
+                (true, true) => stats.page_images += 1,
```

Comment:
Is "page" the right nomenclature here, or should we use "block" or "data"? This seems fairly inconsistent in general, but I think we informally tend to use page.

Reply:
page is kind of a an inaccurate name for the values in our KV store, but it's baked into `pageserver` :-/

I'm fine with whatever seems natural to you

Reply:
Went with `data`, to differentiate metadata pages from data pages.